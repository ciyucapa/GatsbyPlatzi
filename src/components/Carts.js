import React from "react";
import {Link} from "gatsby";
import PropTypes from "prop-types"

import {Button, StyledCart} from'../styles/components'

const Carts = (props) => (
    <StyledCart>
        <h2>Carrito de compras</h2>
        <table>
            <tbody>
                <tr>
                    <th>Product</th>
                    <th>Precio</th>
                    <th>Cantidad</th>
                    <th>Total</th>
                </tr>
                {props.cart.map(swag => (
                    <tr key={swag.id}>
                        <td>
                            <img src={swag.image} alt={swag.name}/>
                        </td>
                        <td>USD {props.priceFormat(swag.price)}</td>
                        <td>{swag.qty}</td>
                        <td>USD {props.priceFormat(swag.qty * swag.price)}</td>
                    </tr>
                ))}
            </tbody>
        </table>
        <nav>
            <div>
                <h3>Subtotal:</h3>
                <small>USD {props.priceFormat(props.total)}</small>
            </div>
            <div>
                <Link to='/'>
                    <Button type='outline'>Volver</Button>
                </Link>
                <Button 
                    onClick={props.handleBuy} 
                    disabled={props.cart.length === 0} 
                    type='outline'
                >
                    Comprar
                </Button>
            </div>
        </nav>
    </StyledCart>
);

Carts.propTypes = {
    total: PropTypes.number,
    cart: PropTypes.arrayOf(PropTypes.object),
    handleBuy: PropTypes.func,
    priceFormat: PropTypes.func,
}
  
Carts.defaultProps = {
    cart: [],
    total: 0,
    handleBuy: () => {},
    priceFormat: () => {},
}

export default Carts;