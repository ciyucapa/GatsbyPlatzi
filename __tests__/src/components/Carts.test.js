import React from "react"
import renderer from "react-test-renderer"
import Carts from '../../../src/components/Carts'

describe("Carts", () => {
    it("renders correctly", () => {
      const tree = renderer
        .create(<Carts 
                   priceFormat={() => 3500}
                   handleBuy={() => 1}
                   cart={[{id: 3500, image: 'http://test.com/image', price: 3500, qty:1}]}
                   total={3500} 
                />)
        .toJSON()
      expect(tree).toMatchSnapshot()
    });
});