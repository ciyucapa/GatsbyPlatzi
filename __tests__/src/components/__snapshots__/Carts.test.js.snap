// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carts renders correctly 1`] = `
<div
  className="sc-ifAKCX kThCum"
>
  <h2>
    Carrito de compras
  </h2>
  <table>
    <tbody>
      <tr>
        <th>
          Product
        </th>
        <th>
          Precio
        </th>
        <th>
          Cantidad
        </th>
        <th>
          Total
        </th>
      </tr>
      <tr>
        <td>
          <img
            src="http://test.com/image"
          />
        </td>
        <td>
          USD 
          3500
        </td>
        <td>
          1
        </td>
        <td>
          USD 
          3500
        </td>
      </tr>
    </tbody>
  </table>
  <nav>
    <div>
      <h3>
        Subtotal:
      </h3>
      <small>
        USD 
        3500
      </small>
    </div>
    <div>
      <a
        href="/"
      >
        <button
          className="sc-gqjmRU kQlgie"
          type="outline"
        >
          Volver
        </button>
      </a>
      <button
        className="sc-gqjmRU kQlgie"
        disabled={false}
        onClick={[Function]}
        type="outline"
      >
        Comprar
      </button>
    </div>
  </nav>
</div>
`;
